tests/test_memory.py

```diff
@@ -0,0 +1,336 @@
+# Includes code from
+# https://github.com/joblib/joblib/blob/6836640abba55611d6e57f20338ea54b3e27f296/joblib/test/test_hashing.py
+
+import gc
+import itertools
+import pickle
+import sys
+import time
+from concurrent.futures import ProcessPoolExecutor
+
+import numpy as np
+import pandas as pd
+import pytest
+from xxhash import xxh128
+
+from pycaret.internal.memory import fast_hash as hash
+
+
+@pytest.fixture(scope="function")
+def three_np_arrays():
+    rnd = np.random.RandomState(0)
+    arr1 = rnd.random_sample((10, 10))
+    arr2 = arr1.copy()
+    arr3 = arr2.copy()
+    arr3[0] += 1
+    return arr1, arr2, arr3
+
+
+# Helper functions for the tests
+def time_func(func, *args):
+    """Time function func on *args."""
+    times = list()
+    for _ in range(3):
+        t1 = time.time()
+        func(*args)
+        times.append(time.time() - t1)
+    return min(times)
+
+
+def relative_time(func1, func2, *args):
+    """Return the relative time between func1 and func2 applied on
+    *args.
+    """
+    time_func1 = time_func(func1, *args)
+    time_func2 = time_func(func2, *args)
+    if time_func1 + time_func2 == 0:
+        return 0
+    relative_diff = 0.5 * (abs(time_func1 - time_func2) / (time_func1 + time_func2))
+    return relative_diff
+
+
+@pytest.mark.parametrize("dtype", ("int", "category", "object"))
+def test_pandas_dataframe(dtype):
```

Comment:
why are we testing this? Isn't this outside of pycaret's scope?

Reply:
It is necessary as we have added new hashing logic that is not tested in joblib! The test suite is minimal, and benchmarks can be turned off if needed.

Reply:
alright. accepted